keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trainer for Word Piece Tokenizer."""
+
+from collections import Counter
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+from tensorflow_text.tools.wordpiece_vocab import (
+    wordpiece_tokenizer_learner_lib as learner,
+)
+
+from keras_nlp.tokenizers.word_piece_tokenizer import PUNCTUATION_REGEX
+from keras_nlp.tokenizers.word_piece_tokenizer import (
+    WHITESPACE_AND_PUNCTUATION_REGEX,
+)
+
+
+def compute_word_piece_vocabulary(
+    data,
+    vocabulary_size: int,
+    split=True,
+    lowercase: bool = True,
+    strip_accents: bool = True,
+    vocabulary_output_file=None,
+    reserved_tokens=["[PAD]", "[CLS]", "[SEP]", "[UNK]", "[MASK]"],
+    suffix_indicator="##",
+):
+    """A utility to train a Word Piece vocabulary.
+
+    This function can be used to train a Word Piece vocabulary from an input
+    dataset, or a list of text strings.
+
+    Args:
+        data: A tf.data.Dataset, or a list of text strings.
```

Comment:
I thought we were going to still support file loading, just not with `split=False`?

Reply:
Oh yep, I just uploaded what I had yesterday, haven't made the edits from the most recent discussion yet, will change it now.